de.properties.tmpl

```diff
@@ -0,0 +1,95 @@
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.logout = {{ $v }}/secured/logout{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.de-data-mgmt.base = {{ $v }}/secured/filesystem/{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.admin.de-data-mgmt.base = {{ $v }}/admin/filesystem/{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.file-io.base.secured = {{ $v }}/secured/fileio/{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.file-io.file-upload = {{ $v }}/fileio/upload{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.file-io.secured.file-upload = {{ $v }}/secured/fileio/upload{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.buggalo.baseUrl = {{ $v }}/secured/tree-viewer-urls{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.toolRequests = {{ $v }}/tool-requests{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.toolRequests.statusCodes = {{ $v }}/tool-requests/status-codes{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.admin.toolRequests = {{ $v }}/admin/tool-requests{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.permIdRequests = {{ $v }}/permanent-id-requests{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.permIdRequests.statusCodes = {{ $v }}/permanent-id-requests/status-codes{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.admin.permIdRequests = {{ $v }}/admin/permanent-id-requests{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.apps = {{ $v }}/apps{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.apps.ids = {{ $v }}/apps/ids{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.apps.categories = {{ $v }}/apps/categories{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.apps.hierarchies = {{ $v }}/apps/hierarchies{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.apps.elements = {{ $v }}/apps/elements{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.apps.elements.tools = {{ $v }}/apps/elements/tools{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.apps.elements.infoTypes = {{ $v }}/apps/elements/info-types{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.apps.elements.dataSources = {{ $v }}/apps/elements/data-sources{{ end }}
+{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.apps.argPreview = {{ $v }}/apps/arg-preview{{ end }}
```

Comment:
So just to clarify in text the bit I was poorly explaining in person: since `with` acts as an implicit `if` statement, if for some reason the terrain base wasn't in consul, this line would end up empty. Since from what you said the UI requires each thing to be there, it might make more sense to change:

```
{{ with $v := (key (printf "%s/terrain/base" $base)) }}org.iplantc.services.apps.argPreview = {{ $v }}/apps/arg-preview{{ end }}
```

to 

```
org.iplantc.services.apps.argPreview = {{ key (printf "%s/terrain/base" $base) }}/apps/arg-preview
```

Both will probably technically be wrong in the case of a missing variable, but the latter is probably actually easier to read in a case like this where there's no defaults in play to make it better to exclude the property entirely when it's not set.